File: app/src/main/java/com/m6world/tari_labs/api/_interactors/AuthInteractor.java
package com.m6world.tari_labs.api._interactors;

import com.m6world.tari_labs.api._contracts.IAuthInteractor;
import com.m6world.tari_labs.api.models.AuthResponse;
import com.m6world.tari_labs.api.models.TokenRequest;
import com.m6world.tari_labs.api.models.RefreshRequest;
import io.reactivex.Observable;

public class AuthInteractor implements IAuthInteractor {
    PostNetworkInteractor postNetworkInteractor = new PostNetworkInteractor();

    private final String AUTH_TOKEN_ENDPOINT = "auth/token";
    private final String AUTH_TOKEN_REFRESH_ENDPOINT = "auth/token/refresh";

    public Observable<AuthResponse> getAuthToken(String email, String password) {
        TokenRequest request = new TokenRequest(email, password);
        return postNetworkInteractor.createObservable(AUTH_TOKEN_ENDPOINT, request, AuthResponse.class);
    }

    public Observable<AuthResponse> getAuthRefreshToken(String accessToken, String refreshToken) {
        RefreshRequest request = new RefreshRequest(accessToken, refreshToken);
        return postNetworkInteractor.createObservable(AUTH_TOKEN_REFRESH_ENDPOINT, request, AuthResponse.class);
    }
}

File: app/src/main/java/com/m6world/tari_labs/api/interactors/AuthInteractor.kt
package com.m6world.tari_labs.api.interactors

import com.m6world.tari_labs.api.contracts.IAuthInteractor
import com.m6world.tari_labs.api.models.AuthResponse
import com.m6world.tari_labs.api.models.TokenRequest
import com.m6world.tari_labs.api.models.RefreshRequest
import io.reactivex.Observable

class AuthInteractor : IAuthInteractor {
    internal var postNetworkInteractor = PostNetworkInteractor()

    private val AUTH_TOKEN_ENDPOINT = "auth/token"
    private val AUTH_TOKEN_REFRESH_ENDPOINT = "auth/token/refresh"

    override fun getAuthToken(email: String, password: String): Observable<AuthResponse> {
        val request = TokenRequest(email, password)
        return postNetworkInteractor.createObservable(AUTH_TOKEN_ENDPOINT, request, AuthResponse::class.java)
    }

    override fun getAuthRefreshToken(accessToken: String, refreshToken: String): Observable<AuthResponse> {
        val request = RefreshRequest(accessToken, refreshToken)
        return postNetworkInteractor.createObservable(AUTH_TOKEN_REFRESH_ENDPOINT, request, AuthResponse::class.java)
    }
}